src/tests/domain_resolution_order-tests.c

```diff
@@ -0,0 +1,227 @@
+/*
+    Authors:
+        Fabiano Fidêncio <fidencio@redhat.com>
+
+    Copyright (C) 2018 Red Hat
+
+    SSSD tests: Domain Resolution Order tests
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <stdlib.h>
+#include <check.h>
```

Comment:
Is there a reason for using the old check test-framework and not cmocka?

Reply:
Well, no. No specific reason.

The few things that crossed my mind while writing those is that:
- We don't need any mock to run those tests;
- We don't run cmocka tests in a few platforms (EL)

Would you prefer that I migrate those to cmocka even if I don't use anything from there?

Reply:
I think it is a fair point that we do not run the cmocka tests on all platforms because cmocka is not available, so please keep the test as it is.